sympy/combinatorics/rewritingsystem_fsm.py

```diff
@@ -0,0 +1,63 @@
+class State(object):
+    '''
+    A representation of a state managed by a ``StateMachine``.
+
+    Attributes:
+        name (instance of FreeGroupElement or string) -- State name which is also assigned to the Machine.
+        transisitons (OrderedDict) -- Represents all the transitions of the state object.
+        state_type (string) -- Denotes the type (accept/start/dead) of the state.
+        rh_rule (instance of FreeGroupElement) -- right hand rule for dead state.
+    '''
+
+    def __init__(self, name, state_type=None, rh_rule=None):
+        self.name = name
+        self.transitions = {}
+        self.state_type = state_type
+        self.rh_rule = rh_rule
+        if self.state_type == 's':
+            self.state_type = "start"
+        elif self.state_type == 'a':
+            self.state_type = "accept"
+        elif self.state_type == 'd':
+            self.state_type = "dead"
+
+    def add_transition(self, letter, state):
+        '''
+        Add a transition from the current state to a new state.
+
+        Keyword Arguments:
+            letter -- The alphabet element the current state reads to make the state transition.
+            state -- This will be an instance of the State object which represents a new state after in the transition after the alphabet is read.
+
+        '''
+        self.transitions[letter] = state
```

Comment:
I feel that a state should have a state machine associated with it, because it doesn't make sense to have a state without a machine. Maybe as a `state_machine` attribute? Then if you also add an `alphabet` attribute to `StateMachine`, you could check here whether `letter` is in `state_machine.alphabet`.